spoolman/api/v1/qr.py

```diff
@@ -0,0 +1,63 @@
+"""QR-Code related endpoints."""
+
+# ruff: noqa: D103
+
+import io
+import logging
+from typing import Annotated
+
+import qrcode
+from fastapi import APIRouter, Response
+from fastapi.responses import StreamingResponse
+from pydantic import BaseModel
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter(
+    prefix="",
+    tags=["qrcode"],
+)
+
+
+class QRRequestBody(BaseModel):
+    """The request body for the QR code endpoint."""
+
+    data: Annotated[str, "The data to encode into the QR code."]
+    box_size: Annotated[int, "The size of the boxes in the QR code."] = 10
+    border: Annotated[int, "The size of the border around the QR code."] = 2
+
+
+@router.post(
```

Comment:
Don't see a reason have the backend generate the QR code.
You can use the exiting QRCode from antd to create a download-able image.
See https://ant.design/components/qr-code#qr-code-demo-download

Reply:
I just like to have the backend, do most of the work. Honestly I would have liked to move the CSV generation to the backend as well, but couldn't quite get it to work properly.

The main reason is, I don't know how the JS libraries will behave in different browsers / different OS's, but the backend will always work the same way.
And maybe someday we will see a native Spoolman mobile App, which would require the same generations ^^

Reply:
The big downside to having the backend do this is that it opens up for other parties to start using it, and then it becomes increasingly difficult to make changes to it without breaking backwards compatibility. I would also prefer for the frontend to do this since it's fully capable of doing it. It also puts less stress on the backend server.

Styling etc can be tedious with different browsers and OSes, but that shouldn't apply to generating images using some library.